src/components/enketo-iframe.vue

```diff
@@ -47,24 +47,43 @@ const { t } = useI18n();
 const invalidProps = computed(() => {
   // actionType is '' for public-link
   const validActionTypes = ['offline', 'edit', 'new', 'preview', ''];
-  if (!props.enketoId) return true;
   if (!validActionTypes.includes(props.actionType)) return true;
   if (props.actionType === 'edit' && !props.instanceId) return true;
   return false;
 });
 
 const redirectUrl = computed(() => route.query.return_url);
 
-const enketoSrc = computed(() => {
-  let prefix = '/enketo-passthrough';
-  const { return_url: _, ...query } = route.query;
-
-  query.parentWindowOrigin = window.location.origin;
-
+const lastSubmitted = (enketoOnceId) => {
+  const iframe = document.createElement('iframe');
+  iframe.src = '/-/single/check-submitted';
+  iframe.style.display = 'none';
+
+  return new Promise((resolve) => {
+    iframe.onload = () => {
```

Comment:
This might be done in another PR, but what do you think about showing a loading message while `enketoSrc.value == null`? Since checking the `enketoOnceId` cookie is an async process, I think there's a possibility that the page could be completely blank while the cookie iframe is loading.

Reply:
Related idea: now that `enketoSrc.value` can be nullish, what do you think about checking for that possibility in the template? I.e., changing the `v-else` to `v-else-if`. I don't think it was possible before for `enketoSrc.value` to be nullish, at least as long as `invalidProps.value` was `false`.

```vue
<iframe v-else-if="enketoSrc != null" ... :src="enketoSrc">
```